File: bench/java/Matrix.java
public class Matrix {
    public static void main(String[] args) {
        int n = 500;
        double[] a = new double[n * n];
        double[] b = new double[n * n];
        double[] c = new double[n * n];

        for (int i = 0; i < n * n; i++) {
            a[i] = i % 100;
            b[i] = (i * 7) % 100;
        }

        long start = System.nanoTime();

        // ikj loop order for cache efficiency
        for (int i = 0; i < n; i++) {
            for (int k = 0; k < n; k++) {
                double a_ik = a[i * n + k];
                for (int j = 0; j < n; j++) {
                    c[i * n + j] += a_ik * b[k * n + j];
                }
            }
        }

        double ms = (System.nanoTime() - start) / 1e6;

        System.out.printf("C[0][0]=%.0f C[499][499]=%.0f%n", c[0], c[499 * n + 499]);
        System.out.printf("RESULT:%.3f%n", ms);
    }
}

File: bench/js/matrix.js
const N = 200;
const A = new Float64Array(N * N);
const B = new Float64Array(N * N);
const C = new Float64Array(N * N);

for (let i = 0; i < N * N; i++) {
    A[i] = i % 100;
    B[i] = (i * 7) % 100;
}

const start = process.hrtime.bigint();

for (let i = 0; i < N; i++) {
    for (let k = 0; k < N; k++) {
        const a_ik = A[i * N + k];
        for (let j = 0; j < N; j++) {
            C[i * N + j] += a_ik * B[k * N + j];
        }
    }
}

const ms = Number(process.hrtime.bigint() - start) / 1e6;

console.log(`C[0][0]=${C[0].toFixed(0)} C[${N-1}][${N-1}]=${C[(N-1) * N + (N-1)].toFixed(0)}`);
console.log(`RESULT:${ms.toFixed(3)}`);
